SoftLayer/CLI/hardware/create_options.py

```diff
@@ -22,41 +26,171 @@ def cli(env):
+    if location and prices:
+        raise exceptions.CLIAbort("Please select --prices or --location to get the prices, not both")
+
+    if prices:
+        _preset_prices_table(options['sizes'], tables)
+        _os_prices_table(options['operating_systems'], tables)
+        _port_speed_prices_table(options['port_speeds'], tables)
+        _extras_prices_table(options['extras'], tables)
+    elif location:
+        _preset_prices_table(options['sizes'], tables)
+        location_prices = hardware_manager.get_hardware_item_prices(location)
+        _location_item_prices(location_prices, tables)
+    else:
+        # Presets
+        preset_table = formatting.Table(['Size', 'Value'], title="Sizes")
+        preset_table.sortby = 'Value'
+        preset_table.align = 'l'
+        for size in options['sizes']:
+            preset_table.add_row([size['name'], size['key']])
+        tables.append(preset_table)
+
+        # Operating systems
+        os_table = formatting.Table(['OS', 'Key', 'Reference Code'], title="Operating Systems")
+        os_table.sortby = 'Key'
+        os_table.align = 'l'
+        for operating_system in options['operating_systems']:
+            os_table.add_row([operating_system['name'], operating_system['key'], operating_system['referenceCode']])
+        tables.append(os_table)
+
+        # Port speed
+        port_speed_table = formatting.Table(['Network', 'Speed', 'Key'], title="Network Options")
+        port_speed_table.sortby = 'Speed'
+        port_speed_table.align = 'l'
+        for speed in options['port_speeds']:
+            port_speed_table.add_row([speed['name'], speed['speed'], speed['key']])
+        tables.append(port_speed_table)
+
+        # Extras
+        extras_table = formatting.Table(['Extra Option', 'Value'], title="Extras")
+        extras_table.sortby = 'Value'
+        extras_table.align = 'l'
+        for extra in options['extras']:
+            extras_table.add_row([extra['name'], extra['key']])
+        tables.append(extras_table)
 
     env.fout(formatting.listing(tables, separator='\n'))
+
+
+def _preset_prices_table(sizes, tables):
+    """Shows Server Preset options prices.
+
+    :param [] sizes: List of Hardware Server sizes.
+    :param tables: Table formatting.
+    """
+    preset_prices_table = formatting.Table(['Size', 'Value', 'Hourly', 'Monthly'], title="Sizes Prices")
+    preset_prices_table.sortby = 'Value'
+    preset_prices_table.align = 'l'
+    for size in sizes:
+        preset_prices_table.add_row([size['name'], size['key'], size['hourlyRecurringFee'], size['recurringFee']])
+    tables.append(preset_prices_table)
+
+
+def _os_prices_table(operating_systems, tables):
+    """Shows Server Operating Systems prices cost and capacity restriction.
+
+    :param [] operating_systems: List of Hardware Server operating systems.
+    :param tables: Table formatting.
+    """
+    os_prices_table = formatting.Table(['OS Key', 'Hourly', 'Monthly', 'capacityRestrictionMaximum',
```

Comment:
To make this easier to read, lets make the capacity restriction part a bit more condensed for all tables with this field

```
:................................................................................:
: Operating Systems Prices                                                       :
:.............................................:........:.........:...............:
: OS Key                                      : Hourly : Monthly : Restriction   : 
:.............................................:........:.........:...............:
: OS_CENTOS_6_X_32_BIT                        : 0      : 0       : None          :
: OS_WINDOWS_2008_FULL_WEB_64_BIT_R2_SP1      : -      : 11      : 1-1 PROCESSOR :
: OS_WINDOWS_2008_FULL_WEB_64_BIT_R2_SP1      : -      : 22      : 2-2 PROCESSOR :
: OS_WINDOWS_2012_FULL_STD_64_BIT             : .056   : 36      : 2-4 CORE      :
: OS_WINDOWS_2012_FULL_STD_64_BIT             : .112   : 72      : 6-16 CORE     :
:.............................................:........:.........:...............:
```

Reply:
Done.